[MTA 8.2.0] New mta-ops guide (#383)
migtools/mta-documentation: docs/topics/mta-ops/ref_common-migration-pipeline-errors.adoc

:_newdoc-version: 2.18.7
:_template-generated: 2026-07-07
:_mod-docs-content-type: REFERENCE

[id="common-migration-pipeline-errors_{context}"]
= Common migration pipeline errors

[role="_abstract"]
When you encounter pipeline failures, review the common errors tables. This helps you diagnose issues and resume your migration work.

.Export and transform errors
[cols="20%,30%,50%",options="header"]
|===
| Error message | Cause | Solution

| `Forbidden`

> | Missing role-based access control (RBAC) permissions for a resource.

Author: ```suggestion
| You lack role-based access control (RBAC) permissions to export a resource.
```

> | Review the `failures/` directory. Use role impersonation flags.

Author: ```suggestion
| Inspect the `failures/` directory for skipped dependencies. Use role impersonation flags such as `--as` and `--as-group`.
```

| `Patch evaluation error`
| A custom stage contains malformed YAML or missing parent objects.

> | Isolate the stage using `--stage`. Rerun with `--force`.

Reviewer: There is no transform argument `--stage`. Since May, the stage name is a positional argument.

Author: @aufi would that be correct?
```suggestion
| Isolate the stage by passing the directory name as a positional argument. Rerun the command with the `--force` flag.
```

Reviewer: Currently, yes.

|===

.Apply and validate errors
[cols="20%,30%,50%",options="header"]
|===
| Error message | Cause | Solution

| `Dependency order failure`

> | Resources applied from a directory executed out of order.

Author: ```suggestion
| The deployment command executed manifest files out of their required dependency order.
```

> | Run `apply` with the `--ordered` flag.

Author: ```suggestion
| Rerun the `mta-ops apply` command with the `--ordered` flag.
```

| `Unknown resource type`
| A required Custom Resource Definition (CRD) is missing.

> | Install the required Operator on the target cluster.

Author: ```suggestion
| A required Custom Resource Definition (CRD) does not exist on the target cluster.
| Install the required Operator on the target cluster before you deploy the generated manifests.
```
